fuel-core/src/schema/tx.rs

```diff
@@ -266,20 +274,72 @@ impl TxMutation {
         if let Some(utxo_validation) = utxo_validation {
             cfg.utxo_validation = utxo_validation;
         }
-        let tx = FuelTx::from_bytes(&tx.0)?;
-        // make virtual txpool from transactional view
-        let tx_pool = TxPool::new(transaction.deref().clone(), cfg.clone());
-        let receipts = tx_pool.run_tx(tx).await?;
-        Ok(receipts.into_iter().map(Into::into).collect())
+        let mut tx = FuelTx::from_bytes(&tx.0)?;
+        tx.precompute_metadata();
+        let id = tx.id();
+
+        // make executor from transaction database view.
+        let executor = Executor {
+            database: transaction.deref().clone(),
+            config: cfg.clone(),
+        };
+        executor.submit_txs(vec![Arc::new(tx)]).await?;
+        // get receipts from db transaction
+        let receipts = Storage::<FuelBytes32, Vec<FuelReceipt>>::get(transaction.deref(), &id)?
+            .unwrap_or_default();
+        Ok(receipts.iter().map(Into::into).collect())
     }
 
     /// Submits transaction to the txpool
     async fn submit(&self, ctx: &Context<'_>, tx: HexString) -> async_graphql::Result<Transaction> {
-        let tx_pool = ctx.data::<Arc<TxPool>>().unwrap();
-        let tx = FuelTx::from_bytes(&tx.0)?;
-        tx_pool.submit_tx(tx.clone()).await?;
-        let tx = Transaction(tx);
+        let db = ctx.data_unchecked::<Database>();
+        let txpool = ctx.data::<Arc<TxPoolService>>().unwrap();
```

Comment:
why not the same approach: `let txpool = ctx.data_unchecked::<Arc<TxPoolService>>();` ?